modules/oe_whitelabel_paragraphs/oe_whitelabel_paragraphs.module

```diff
@@ -208,7 +208,69 @@ function oe_whitelabel_preprocess_paragraph__oe_text_feature_media(array &$varia
   }
 
   $cacheability->applyTo($variables);
+}
+
+/**
+ * Implements hook_preprocess_paragraph() for paragraph--oe-av-media.html.twig.
+ */
+function oe_whitelabel_preprocess_paragraph__oe_av_media(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  // Caches are handled by the formatter usually. Since we are not rendering
+  // the original render arrays, we need to propagate our caches to the
+  // paragraph template.
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
+
+  $cacheability->applyTo($variables);
+
+  // Bail out if there is no media present.
+  if ($paragraph->get('field_oe_media')->isEmpty()) {
+    return;
+  }
+
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = $paragraph->get('field_oe_media')->entity;
+  if (!$media instanceof MediaInterface) {
+    // The media entity is not available anymore, bail out.
+    return;
+  }
+
+  // Retrieve the correct media translation.
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = \Drupal::service('entity.repository')->getTranslationFromContext($media, $paragraph->language()->getId());
+
+  $cacheability->addCacheableDependency($media);
+
+  // Run access checks on the media entity.
+  $access = $media->access('view', $variables['user'], TRUE);
+  $cacheability->addCacheableDependency($access);
+  if (!$access->isAllowed()) {
+    $cacheability->applyTo($variables);
+    return;
+  }
+
+  // Get the media source.
+  $source = $media->getSource();
+
+  $is_image = $source instanceof MediaAvPortalPhotoSource || $source instanceof Image;
+  $is_video = $source instanceof MediaAvPortalVideoSource || $source instanceof OEmbed || $source instanceof Iframe;
 
+  // If it's not an image and not a video, bail out.
+  if (!$is_image && !$is_video) {
+    $cacheability->applyTo($variables);
+    return;
+  }
+
+  if ($is_image) {
+    $thumbnail = $media->get('thumbnail')->first();
+    $variables['image'] = ImageValueObject::fromStyledImageItem($thumbnail, 'oe_bootstrap_theme_medium_no_crop');
+  }
+  elseif ($is_video) {
+    _oe_whitelabel_featured_media_set_embedded_media($variables, $media, $cacheability, $source);
+  }
+
+  $cacheability->applyTo($variables);
```

Comment:
Most of this is duplicated, shall we have a separate method?

Reply:
I've added a separate method, did notice big parts of this code also in other submodules, was thinking, often projects copy this code, would be even more reusable for use and them if we create a class where we collect most of this code, what do you think?